collect_app/src/androidTest/java/org/odk/collect/android/support/ResetStateRule.kt

```diff
@@ -0,0 +1,71 @@
+package org.odk.collect.android.support
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import org.apache.commons.io.FileUtils
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+import org.odk.collect.android.TestSettingsProvider.getSettingsProvider
+import org.odk.collect.android.database.DatabaseConnection.Companion.closeAll
+import org.odk.collect.android.injection.DaggerUtils
+import org.odk.collect.android.injection.config.AppDependencyModule
+import org.odk.collect.android.storage.StoragePathProvider
+import org.odk.collect.android.utilities.MultiClickGuard
+import org.odk.collect.android.views.DecoratedBarcodeView
+import org.odk.collect.androidshared.data.getState
+import org.odk.collect.androidshared.ui.ToastUtils.recordToasts
+import java.io.File
+import java.io.IOException
+
+class ResetStateRule @JvmOverloads constructor(private val appDependencyModule: AppDependencyModule? = null) : TestRule {
+
+    private val settingsProvider = getSettingsProvider()
+
+    override fun apply(base: Statement, description: Description): Statement = ResetStateStatement(base)
+
+    private inner class ResetStateStatement(private val base: Statement) : Statement() {
```

Comment:
I'm guessing this is probably what Kotlin conversion gave you, but you can get rid of the `inner` here by moving the helper methods into `ResetStateStatement`. Then you can just move `ResetStateStatement` to the top level as a `private` class in the same file. Definitely an advantage of Kotlin!

Reply:
Done.